src/blr/cli/tagfastq.py

```diff
@@ -88,6 +89,22 @@ def main(args):
     logger.info("Finished")
 
 
+def search_bc(iterator: iter, header: str, cache: dict, maxiter: int = 10):
+    # Check it header is stored in cache. If not move forward on step in iterator at look again.
+    iteration = 0
+    while header not in cache and iteration < maxiter:
+        iteration += 1
+        # Progress iterator
```

Comment:
Initially I read this as having something to do with a progress report ... I think "advance iterator" would be more customary. And consider whether the comment is necessary at all. The `next(iterator)` already describes exactly that.